src/Raven.Server/Documents/Handlers/BatchHandler.cs

```diff
@@ -868,6 +871,21 @@ public class MergedBatchCommand : TransactionMergedCommand, IDisposable
 
             public bool IsClusterTransaction;
 
+            public long GetMaxClusterTransactionIndex()
```

Comment:
Isn't this already handled by the client, when we send the header?
Basically, whenever we observe a `RAFT-???` we need to update, no?

Reply:
Here we get the cluster tx document by replication from a different node, so we can't rely on the raft index of the local node, it can be not in sync with the document. 
On the client you can't do that, because you need to work with the guid, and you don't know it.

I think that another way of solving it, might be to extract the raft index on the server during the `Load`.

Reply:
You are in the batch handler, getting documents from the _client_. So obvious you have them.